src/components/TodoList/TodoList.tsx

```diff
@@ -12,25 +12,27 @@ export const TodoList: React.FC = () => {
   const todos = useAppSelector(state => state.todos);
 
   const filteredTodos = () => {
-    let copiedTodos = [...todos].filter(todo => {
-      switch (status) {
-        case 'all':
-          return todos;
-        case 'active':
-          return !todo.completed;
-        case 'completed':
-          return todo.completed;
-        default:
-          return todo;
-      }
-    });
+    const visibleTodos = [...todos];
 
-    return copiedTodos.filter(todo =>
-      todo.title.toLowerCase().includes(query.trim().toLowerCase()),
-    );
+    return visibleTodos
+      .filter(todo =>
+        todo.title.toLowerCase().includes(query.trim().toLowerCase()),
+      )
+      .filter(todo => {
+        switch (status) {
+          case 'active':
+            return !todo.completed;
+
+          case 'completed':
+            return todo.completed;
+
+          default:
+            return todos;
```

Comment:
todos here will be inexplictly transformed into boolean (since it's array `!!todos` always will be `true`), but it's better to explicitly write the correct boolean value: `return true;`

> Array.fliter callback should return boolean - you have case when you return an array and an object